File: restack_up.ts
import * as dotenv from "dotenv";
import RestackCloud from "@restackio/restack-sdk-cloud-ts";
dotenv.config();

const main = async () => {
  const restackCloudClient = new RestackCloud(
    process.env.RESTACK_SDK_TOKEN as string
  );

  const frontendApp = {
    name: "my-ai-application", // application name is used as a persistent identifier. Changing your application name will result in creating a new application on the restack console. The uniqueness is at the stack level. An application can share name as long as they are not in the same stack.
    dockerFilePath: "Dockerfile",
    environmentVariables: [
      {
        name: "TEST_VARIABLE",
        value: "test_2",
      },
    ],
  };

  await restackCloudClient.stack({
    name: "production environment",
    previewEnabled: false,
    applications: [frontendApp],
  });

  await restackCloudClient.up();
};

main();

File: restack_up.py
import os
from restack_sdk_cloud import RestackCloud

async def main():
    # Initialize the RestackCloud client with the SDK token from environment variables
    restack_cloud_client = RestackCloud(os.getenv('RESTACK_SDK_TOKEN'))

    # Define the backend application configuration
    backend_app = {
        'name': 'backend-python',
        'dockerFilePath': 'backend/Dockerfile',
        'dockerBuildContext': 'backend',
    }

    # Define the frontend application configuration
    frontend_app = {
        'name': 'frontend-python',
        'dockerFilePath': 'frontend/Dockerfile',
        'dockerBuildContext': 'frontend',
        'environmentVariables': [
            {
                'name': 'NEXT_PUBLIC_API_HOSTNAME',
                'linkTo': backend_app['name'],
            },
        ],
    }

    # Configure the stack with the applications
    await restack_cloud_client.stack({
        'name': 'development environment python',
        'previewEnabled': False,
        'applications': [backend_app, frontend_app],
    })

    # Deploy the stack
    await restack_cloud_client.up()

# Run the main function
if __name__ == "__main__":
    import asyncio
    asyncio.run(main())
